(window.webpackJsonp=window.webpackJsonp||[]).push([[31],{627:function(t,e,r){"use strict";r.r(e);var a=r(6),s=Object(a.a)({},(function(){var t=this,e=t.$createElement,r=t._self._c||e;return r("ContentSlotsDistributor",{attrs:{"slot-key":t.$parent.slotKey}},[r("p",[t._v("记录一下今天遇到的流水线的 Bug，改了老半天，最后才发现问题所在！")]),t._v(" "),r("h2",{attrs:{id:"问题复现"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#问题复现"}},[t._v("#")]),t._v(" 问题复现")]),t._v(" "),r("p",[r("img",{attrs:{src:"https://picbed.kimyang.cn/202109050806334.jpg",alt:"image-20200627001641215"}})]),t._v(" "),r("p",[t._v("报错: "),r("code",[t._v("error Command failed with exit code 137")])]),t._v(" "),r("h2",{attrs:{id:"环境说明"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#环境说明"}},[t._v("#")]),t._v(" 环境说明")]),t._v(" "),r("p",[t._v("首先说明报错环境，大致如下：")]),t._v(" "),r("p",[r("img",{attrs:{src:"https://picbed.kimyang.cn/202109050806335.jpg",alt:"博客插图"}})]),t._v(" "),r("h2",{attrs:{id:"原因及解决思路"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#原因及解决思路"}},[t._v("#")]),t._v(" 原因及解决思路")]),t._v(" "),r("p",[t._v("引起这个报错的具体原因其实是 Drone 编译项目时需要的内存容量超过了 Docker 默认分配的 512mb 容量，一句话就是："),r("strong",[t._v("容器内存爆掉了")]),t._v("！！所以解决办法就是扩大 Drone 内存！")]),t._v(" "),r("p",[t._v("如果系统内存主够的话，直接将容器内存扩大就好，具体方法可以参考"),r("a",{attrs:{href:"https://success.docker.com/article/what-causes-a-container-to-exit-with-code-137",target:"_blank",rel:"noopener noreferrer"}},[t._v("此文"),r("OutboundLink")],1),t._v("设定。")]),t._v(" "),r("p",[t._v("但是如果你像我一样用的云服务器，总共内存就只有可怜 🥺 的 2GB 大小，那你可能只能考虑容器添加 SwapFile,来扩充内存！这是一个思路，因为我用的另一位大佬的开发服务器环境，所以给大佬反馈了这个问题之后，大佬自己解决了，这个思路也是大佬说的。具体解决方案，我没有尝试过，有了解的同学可以补充到评论区，感谢 🙏 ～")]),t._v(" "),r("h2",{attrs:{id:"反思及总结"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#反思及总结"}},[t._v("#")]),t._v(" 反思及总结")]),t._v(" "),r("p",[t._v("其实一开始 CI 报错之后我并没有注意到是 Docker 的问题，毕竟 Drone 确实是个可以正常运行的，只是编译的过程中出现错误。而且在编译的过程中 yarn 警告了项目的许多依赖库 outdated ，于是我就以为是依赖库过时导致编译失败。")]),t._v(" "),r("p",[r("img",{attrs:{src:"https://picbed.kimyang.cn/202109050806336.jpg",alt:"image-20200627004226901"}})]),t._v(" "),r("p",[t._v("经过一下午升级相关依赖库及重写部分调用相关 API 的代码，最终尝试在本地编译，编译过程终于没有了警告 ⚠️！！Yes🎉🎉！但是进入流水线之后依旧报错，这时候我才注意到了这个 137 错误代码！！经过一番查询之后这个错误代码是和 Drone 容器内存相关的，于是把问题反馈给了大佬！最后大佬通过添加 swapfile 修复了这个问题！🎉")]),t._v(" "),r("p",[t._v("其实这个问题一个月前便出现了，只是当时自己懒，且很神奇的是当时，第一遍流水线跑不通，重启一次就能跑通！于是懒懒的自己就这样得过且过了！经过一个月，这些没有收敛的问题终究发散到了不得不收拾的地步。其实要是早早修复相关 BUG，今天也不会因为 Outdated 问题 被迷惑了老半天，浪费很多时间最后才查到具体的原因！")]),t._v(" "),r("p",[r("img",{attrs:{src:"https://picbed.kimyang.cn/202109050806337.jpg",alt:"image-20200627004447532"}})]),t._v(" "),r("p",[t._v("总结："),r("strong",[t._v("已经发散的问题如果不及时解决，一定不可能自动收敛，只会引起更多发散的问题！")])]),t._v(" "),r("h2",{attrs:{id:"参考文章"}},[r("a",{staticClass:"header-anchor",attrs:{href:"#参考文章"}},[t._v("#")]),t._v(" 参考文章")]),t._v(" "),r("p",[r("a",{attrs:{href:"https://blog.csdn.net/yuzhongzi81/article/details/51900187",target:"_blank",rel:"noopener noreferrer"}},[t._v("文章 1"),r("OutboundLink")],1)]),t._v(" "),r("p",[r("a",{attrs:{href:"https://success.docker.com/article/what-causes-a-container-to-exit-with-code-137",target:"_blank",rel:"noopener noreferrer"}},[t._v("文章 2"),r("OutboundLink")],1)])])}),[],!1,null,null,null);e.default=s.exports}}]);